### PR TITLE
HIVE: Fix the Hive version checking in MetastoreLock logical error

### PR DESCRIPTION
**Pull request by Paddy0523**
fix the #8440
PTAL.@pvary

### CODE REVIEW

**Comment by pvary** [comment]
@Paddy0523: I think, as a result of this bug, the `fndLock` method was never called in case of the Hive version being higher than 2. Am I right here?

After the fix it will be correctly called.
Could we add assertion(s) to existing tests, that the `fibdLocks` is called when expected too.

Thanks for taking care of this!

**Comment by Paddy0523** [comment]
> Thanks for taking care of this!
@pvary 
You are right.
Thanks for the guidance, I will try to add assertions

**Comment by Paddy0523** [comment]
@pvary
have added the assertion, please look again

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommitLocks.java [hunk 1]
```diff
@@ -226,6 +230,7 @@ public void testLockAcquisitionAfterFailedAndFoundLock() throws TException, Inte
 
     spyOps.doCommit(metadataV2, metadataV1);
 
+    verify(spyClient).showLocks(any()); // Make sure HiveLock's findLock method is called
```
Comment (pvary) [inline hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommitLocks.java:233]: Could we please add `times(1)`, as it is a bit more specific what we expect:
```
verify(spyClient, times(1)).showLocks(any())
```

**Comment by pvary** [comment]
@Paddy0523: Thanks for the fix! 2 small nits, and I think we are good to go.

CC: @RussellSpitzer, @szehon-ho - you might want to know about this fix, as Hive.showLocks codepath was not used previously, and this fixes the issue.

**Review by pvary — APPROVED** [review]

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommitLocks.java (hunk 1 above)
Comment (Paddy0523) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommitLocks.java]: done

**Pull request merged**